### PR TITLE
chore: address Rust 1.75 clippy lints

### PR DESCRIPTION
**Pull request by Chethan-rao**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [x] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This PR addresses the clippy lints occurring due to new rust version 1.75

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by SanchithHegde — DISMISSED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Review by prajjwalkumar17 — DISMISSED** [review]

**Review by pixincreate — DISMISSED** [review]

File: crates/common_utils/src/request.rs [hunk 1]
```diff
@@ -138,7 +138,7 @@ impl RequestBuilder {
     }
 
     pub fn headers(mut self, headers: Vec<(String, Maskable<String>)>) -> Self {
-        let mut h = headers.into_iter().map(|(h, v)| (h, v));
+        let mut h = headers.into_iter();
```
Comment (vspecky) [inline crates/common_utils/src/request.rs:141]: You probably don't even need this. You can pass just `h` to the extend method.

File: crates/router/src/connector/checkout/transformers.rs [hunk 2]
```diff
@@ -1026,10 +1026,7 @@ impl utils::MultipleCaptureSyncResponse for Box<PaymentsResponse> {
         self.status == CheckoutPaymentStatus::Captured
     }
     fn get_amount_captured(&self) -> Option<i64> {
-        match self.amount {
-            Some(amount) => amount.try_into().ok(),
-            None => None,
-        }
+        self.amount.map(|amount| amount.into())
```
Comment (vspecky) [inline crates/router/src/connector/checkout/transformers.rs:1029]:
```suggestion
        self.amount.map(Into::into)
```

File: crates/router/src/core/payments.rs [hunk 3]
```diff
@@ -2160,7 +2160,6 @@ pub async fn apply_filters_on_payments(
         .await
         .to_not_found_response(errors::ApiErrorResponse::PaymentNotFound)?
         .into_iter()
-        .map(|(pi, pa)| (pi, pa))
```
Comment (vspecky) [inline crates/router/src/core/payments.rs:2163]: You probably don't also need the `.into_iter().collect()`

**Review by vspecky — APPROVED** [review]
Just some minor nitpicks. Feel free to ignore these. LGTM otherwise.

**Review by vspecky — APPROVED** [review]

**Pull request merged**